pkg/connector/roles.go

```diff
@@ -93,35 +95,37 @@ func (o *roleBuilder) Entitlements(_ context.Context, resource *v2.Resource, _ *
 }
 
 func (o *roleBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken *pagination.Token) ([]*v2.Grant, string, annotations.Annotations, error) {
+	var grants []*v2.Grant
 	policy, err := o.projectsClient.GetIamPolicy(ctx, &iampb.GetIamPolicyRequest{
 		Resource: fmt.Sprintf("projects/%s", o.bigQueryClient.Project()),
 	})
-	if err != nil {
-		return nil, "", nil, wrapError(err, "failed to get IAM policy")
+	if !isPermissionDenied(ctx, err) {
+		return nil, "", nil, wrapError(err, "listing grants for roles failed")
 	}
 
-	var grants []*v2.Grant
-	for _, binding := range policy.Bindings {
-		if binding.Role != resource.Id.Resource {
-			continue
-		}
-
-		for _, member := range binding.Members {
-			// TODO: handle group bindings
-			if isUser, user := isUser(member); isUser {
-				userResource, err := userResource(user)
-				if err != nil {
-					return nil, "", nil, wrapError(err, "failed to create user resource")
-				}
+	if policy != nil {
```

Comment:
This code would be simpler if it used `if policy == nil` and then early returned.

Reply:
Fixed